wp_api/src/login/url_discovery.rs

```diff
@@ -426,6 +427,33 @@ pub enum AutoDiscoveryAttemptFailure {
         api_root_url: ParsedUrl,
         api_details: WpApiDetails,
     },
+    #[error("Unable to login to {} – the {} plugin might have disabled Application Passwords. Please visit {} to learn more",
+        parsed_site_url,
+        plugin.name,
+        plugin.support_url
+    )]
+    ApplicationPasswordBlockedByPlugin {
+        parsed_site_url: ParsedUrl,
+        api_root_url: ParsedUrl,
+        plugin: KnownApplicationPasswordBlockingPlugin,
+    },
+    #[error("Unable to login to {} – there are multiple installed plugins that might have disabled Application Passwords. Please disable them and try again.",
+        parsed_site_url
+    )]
+    ApplicationPasswordBlockedByMultiplePlugins {
+        parsed_site_url: ParsedUrl,
+        api_root_url: ParsedUrl,
+    },
+    #[error("This site is a local development environment. You'll need to enable application passwords to connect to it with the app.")]
+    SiteIsLocalDevelopmentEnvironment {
+        parsed_site_url: ParsedUrl,
+        api_root_url: ParsedUrl,
+    },
+    #[error("Application Passwords is not enabled for this site – this is likely because we can't establish a secure connection to it. Please add an SSL certificate to this site and try again.")]
+    ApplicationPasswordsDisabledForHttpSite {
+        parsed_site_url: ParsedUrl,
+        api_root_url: ParsedUrl,
+    },
```

Comment:
All these cases seem to be special cases for `ApplicationPasswordsNotSupported`. If I am not mistaken, they don't represent separate states in the api discovery which is what `AutoDiscoveryAttemptFailure` is for.

I think we should move these to a new enum such as `ApplicationPasswordsNotSupportedReason`. That way, we won't need to put `parsed_site_url` and `api_root_url` fields in every variant since it can be retrieved from the parent. It should also significantly clean up the rest of the code from this PR.